File: creational/builder/examples/person-builder.ts
export class PersonBuilder {
    private person = new Person();

    newPerson(): void {
     this.person = new Person()
    }

    setName(name: string): this {
      this.person.name = name;
      return this;
    }

    setAge(age: string): this {
      this.person.age = age;
      return this;
    }


    build(): Person {
      return this.person;
    }
}

const builder = new PersonBuilder();
const person = builder.setName('teste').setAge(2).build();
person.newPerson();
const person = builder.setName('teste2').setAge(2).build();

File: creational/builder/examples/person-builder.java

public class PersonBuilder {
    private Person person;

    public PersonBuilder() {
        this.person = new Person();
    }

    public void newPerson() {
        this.person = new Person();
    }

    public PersonBuilder setName(String name) {
        this.person.setName(name);
        return this;
    }

    public PersonBuilder setAge(int age) {
        this.person.setAge(age);
        return this;
    }

    public Person build() {
        return this.person;
    }
}
